tests/v2/validation/charts/monitoring.go

```diff
@@ -0,0 +1,430 @@
+				},
+				NotifierConfig: config.NotifierConfig{
+					VSendResolved: false,
+				},
+				URL: origin,
+			},
+		},
+	})
+
+	stringifiedAlertConfig := alertConfig.String()
+	encodedStringifiedAlertConfig := base64.StdEncoding.EncodeToString([]byte(stringifiedAlertConfig))
+
+	return encodedStringifiedAlertConfig, nil
+}
+
+// editAlertRoute is a private helper function
+// that edits alert config structure to be used by the webhook receiver.
+func editAlertRoute(alertConfigByte []byte, origin string, originURL *url.URL) (string, error) {
+	alertConfig := config.Config{}
+
+	err := yaml.Unmarshal(alertConfigByte, &alertConfig)
+	if err != nil {
+		return "", fmt.Errorf("Failed to unmarshal alert config: %v", err)
+	}
+
+	alertConfig.Global = &config.GlobalConfig{
+		ResolveTimeout: alertConfig.Global.ResolveTimeout,
+	}
+	alertConfig.Route.Routes = append(alertConfig.Route.Routes, &config.Route{
+		GroupWait:      alertConfig.Route.GroupWait,
+		GroupInterval:  alertConfig.Route.GroupInterval,
+		RepeatInterval: alertConfig.Route.RepeatInterval,
+		Match:          ruleLabel,
+		Receiver:       webhookReceiverDeploymentName,
+	})
+
+	stringifiedAlertConfig := alertConfig.String()
+	encodedStringifiedAlertConfig := base64.StdEncoding.EncodeToString([]byte(stringifiedAlertConfig))
+
+	return encodedStringifiedAlertConfig, nil
+}
+
+// createPrometheusRule is a private helper function
+// that creates a prometheus rule to be used by the webhook receiver.
+func createPrometheusRule(client *rancher.Client, clusterID string) error {
+	ruleName := "webhook-rule-" + namegenerator.RandStringLower(defaultRandStringLength)
+	alertName := "alert-" + namegenerator.RandStringLower(defaultRandStringLength)
+	dynamicClient, err := client.GetDownStreamClusterClient(clusterID)
+	if err != nil {
+		return err
+	}
+
+	prometheusRulesResource := dynamicClient.Resource(prometheusRulesGroupVersionResource).Namespace(charts.RancherMonitoringNamespace)
+	prometheusRule := &monitoringv1.PrometheusRule{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ruleName,
+			Namespace: charts.RancherMonitoringNamespace,
+		},
+		Spec: monitoringv1.PrometheusRuleSpec{
+			Groups: []monitoringv1.RuleGroup{
+				{
+					Name: ruleName,
+					Rules: []monitoringv1.Rule{
+						{
+							Alert:  alertName,
+							Expr:   intstr.IntOrString{Type: intstr.String, StrVal: "vector(0)"},
+							Labels: ruleLabel,
+							For:    "0s",
+						},
+					},
+				},
+			},
+		},
+	}
+	_, err = prometheusRulesResource.Create(context.TODO(), unstructured.MustToUnstructured(prometheusRule), metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// createWebhookReceiverDeployment is a private helper function
+// that creates a service account, cluster role binding, and deployment for webhook receiver.
+func createWebhookReceiverDeployment(client *rancher.Client, clusterID, namespace, deploymentName string) (*appv1.Deployment, error) {
```

Comment:
Same with this function, I think it would work great as an extension. We can ask others looking at the PR to see their opinion.

Reply:
This is partially done. As we discussed, didn't move it to extensions. But enhanced the explanation and changed the helper function name.